src/main/java/org/motechproject/newebodac/mapper/RoleMapper.java

```diff
@@ -0,0 +1,33 @@
+package org.motechproject.newebodac.mapper;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+import org.mapstruct.ReportingPolicy;
+import org.mapstruct.factory.Mappers;
+import org.motechproject.newebodac.domain.security.UserPermission;
+import org.motechproject.newebodac.domain.security.UserRole;
+import org.motechproject.newebodac.dto.RoleDto;
+
+@Mapper(unmappedTargetPolicy = ReportingPolicy.IGNORE)
+public interface RoleMapper extends EntityMapper<RoleDto, UserRole> {
+
+  RoleMapper INSTANCE = Mappers.getMapper(RoleMapper.class);
+
+  @Mapping(target = "id", ignore = true)
+  @Mapping(target = "createDate", ignore = true)
+  @Mapping(target = "updateDate", ignore = true)
+  @Mapping(target = "readonly", ignore = true)
+  void update(RoleDto roleDto, @MappingTarget UserRole role);
+
+  default Set<UserPermission> toUserPermission(UUID id) {
+    return new HashSet<>();
+  }
+
+  default Set<UUID> toUuid(UserPermission id) {
```

Comment:
change the parameter name to userPermission and return UserPermission id here (single id not Set of ids)